src/backend/common/models.py

```diff
@@ -4,14 +4,27 @@
 from django.conf import settings
 from question.models import Solution
 
+
 class User(AbstractUser):
-    avatar = models.URLField(max_length=1024, blank=True, help_text="URL for avatar")
+    avatar = models.URLField(max_length=1024, blank=True,
+                             help_text="URL for avatar")
+    abilityId = models.ForeignKey(
+        settings.QUESTION_ABILITY_MODEL, on_delete=models.CASCADE, null=True)
 
 
 class UserSubmission(models.Model):
-    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, related_name='submission_user')
-    solution = models.ForeignKey(Solution, on_delete=models.CASCADE, related_name='submission_solution')
-    content = models.JSONField()
+    # Representation of the userId (settings.AUTH_USER_MODEL.id)
+    userId = models.BigIntegerField()
+    # Representation of the question id (Question.id)
+    questionId = models.BigIntegerField()
+    # List of category ids, should be validated
+    categoryIds = models.JSONField()
```

Comment:
For simplicity and consistency, we use the singular form, so it should be `category_id`